packages/logging/src/log-escalator.test.ts

```diff
@@ -0,0 +1,310 @@
+            write(msg) {
+                logs.push(msg)
+            },
+        },
+    )
+
+    const wrappedLog = logEscalator(log)
+
+    wrappedLog.trace('Test')
+    wrappedLog.debug('Test')
+    wrappedLog.info('Test')
+    wrappedLog.warn('Test')
+    wrappedLog.error('Test')
+    wrappedLog.fatal('Test')
+
+    expect(logs).toEqual([])
+})
+
+it('wraps a logger', () => {
+    const logs: string[] = []
+    const log = pino(
+        { level: 'trace', timestamp: false, base: {} },
+        {
+            write(msg) {
+                logs.push(msg)
+            },
+        },
+    )
+
+    const wrappedLog = logEscalator(log)
+
+    wrappedLog.trace('Test')
+    wrappedLog.debug('Test')
+    wrappedLog.info('Test')
+    wrappedLog.warn('Test')
+    wrappedLog.error('Test')
+    wrappedLog.fatal('Test')
+    wrappedLog.done()
+
+    expect(logs).toEqual([
+        `{"level":10,"msg":"Test","v":1}
+`,
+        `{"level":20,"msg":"Test","v":1}
+`,
+        `{"level":30,"msg":"Test","v":1}
+`,
+        `{"level":40,"msg":"Test","v":1}
+`,
+        `{"level":50,"msg":"Test","v":1}
+`,
+        `{"level":60,"msg":"Test","v":1}
+`,
+    ])
+})
+
+it('done clears buffer', () => {
+    const logs: string[] = []
+    const log = pino(
+        { level: 'trace', timestamp: false, base: {} },
+        {
+            write(msg) {
+                logs.push(msg)
+            },
+        },
+    )
+
+    const wrappedLog = logEscalator(log)
+
+    wrappedLog.trace('Test')
+    wrappedLog.done()
+    wrappedLog.done()
+
+    expect(logs).toEqual([
+        `{"level":10,"msg":"Test","v":1}
+`,
+    ])
+})
+
+it('does not support creating a child logger', () => {
+    const logs: string[] = []
+    const log = pino(
+        { level: 'info' },
+        {
+            write(msg) {
+                logs.push(msg)
+            },
+        },
+    )
+
+    const wrappedLog = logEscalator(log)
+
+    expect(() => wrappedLog.child({})).toThrowError('')
+})
+
+it('on warn write info', () => {
+    const logs: string[] = []
+    const log = pino(
+        { level: 'warn', timestamp: false, base: {} },
+        {
+            write(msg) {
+                logs.push(msg)
+            },
+        },
+    )
+
+    const wrappedLog = logEscalator(log)
+
+    wrappedLog.trace('Test')
+    wrappedLog.debug('Test')
+    wrappedLog.info('Test')
+    wrappedLog.warn('Test')
+    wrappedLog.done()
+
+    expect(logs).toEqual([
+        `{"level":30,"msg":"Test","v":1}
+`,
+        `{"level":40,"msg":"Test","v":1}
+`,
+    ])
+})
+
+it('on error write debug', () => {
+    const logs: string[] = []
+    const log = pino(
+        { level: 'warn', timestamp: false, base: {} },
+        {
+            write(msg) {
+                logs.push(msg)
+            },
+        },
+    )
+
+    const wrappedLog = logEscalator(log)
+
+    wrappedLog.trace('Test')
+    wrappedLog.debug('Test')
+    wrappedLog.info('Test')
+    wrappedLog.warn('Test')
+    wrappedLog.error('Test')
```

Comment:
A bit of variation in the word test would help make these tests easier to parse for me.

Reply:
Done